### PR TITLE
Add $HADOOP_HOME/hadoop-core-*.jar to the *beginning* of the classpath,

### PR DESCRIPTION
**Pull request by jakemannix**
to make sure that it wins the JarHell battle, in particular to make
sure that "mahout console" works if there is a version mismatch

### CODE REVIEW

File: bin/mahout [hunk 1]
```diff
@@ -164,6 +164,11 @@ for f in $MAHOUT_HOME/examples/target/dependency/*.jar; do
   CLASSPATH=${CLASSPATH}:$f;
 done
 
+# make sure that HADOOP_HOME's hadoop-core jar is first on classpath, to avoid version mismatches
+for f in $HADOOP_HOME/hadoop-core-*.jar; do
```
Comment (sagemintblue) [inline bin/mahout]: quote vars, but avoid quoting globbing part of expression.

File: bin/mahout [hunk 2]
```diff
@@ -164,6 +164,11 @@ for f in $MAHOUT_HOME/examples/target/dependency/*.jar; do
...
+  CLASSPATH=$f:${CLASSPATH};
```
Comment (sagemintblue) [inline bin/mahout]: quotes here too

File: bin/mahout (hunk 1 above)
Comment (jakemannix) [inline bin/mahout]: so, I'm just copying the way they do it above - should we correct all the path / glob handling in this file throughout?

Comment (sagemintblue) [inline bin/mahout]: I'd rather not fix all the deficiencies of this script now, but suffice to say, this bash isn't particularly defensive. On platforms where spaces in paths are more common (Windows) you could get into trouble if path vars are not quoted.

http://mywiki.wooledge.org/BashPitfalls

**Comment by sagemintblue** [comment]
Merging.

**Pull request merged**